### PR TITLE
Add Filter Command Parser

### PR DESCRIPTION
**Pull request by JukeBoxK**
Implementation of the parser for the filter command

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java [hunk 1]
```diff
@@ -109,21 +171,56 @@ private static ArgumentMultimap extractArguments(String argsString, List<PrefixP
...
         Prefix prefix = currentPrefixPosition.getPrefix();
 
-        int valueStartPos = currentPrefixPosition.getStartPosition() + prefix.getPrefix().length();
+        int valueStartPos = currentPrefixPosition.getStartPosition() + prefix.getLength();
+        String value = argsString.substring(valueStartPos, nextPrefixPosition.getStartPosition());
+
+        return value.trim();
+    }
+
+    private static ArgumentMultimap extractFilterArguments(String argsString, List<PrefixPosition> prefixPositions) {
```
Comment (radeon2525) [inline src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java:185]: Maybe you can provide Javadocs for this method to make it easier for developers to understand your logic better

File: src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java [hunk 2]
```diff
@@ -109,21 +171,56 @@ private static ArgumentMultimap extractArguments(String argsString, List<PrefixP
...
+        for (int i = 0; i < prefixPositions.size() - 1; i++) {
+            Prefix argPrefix = prefixPositions.get(i).getPrefix();
+            String argValue = extractFilterArguments(argsString, prefixPositions.get(i), prefixPositions.get(i + 1));
+            argMultimap.put(argPrefix, argValue);
+        }
+
+        return argMultimap;
+    }
+
+    private static String extractFilterArguments(String argsString,
```
Comment (radeon2525) [inline src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java:205]: Same as previous comment, a javadoc would be better suited as these are non-trivial methods

File: src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java (hunk 1 above)
Comment (JukeBoxK) [reply src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java:185]: Done

File: src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java (hunk 2 above)
Comment (JukeBoxK) [reply src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java:222]: Done

**Pull request merged**